logging/src/jvmTest/kotlin/io/spine/logging/JvmLoggerSpec.kt

```diff
@@ -89,6 +92,71 @@ internal class JvmLoggerSpec {
         consoleOutput shouldContain "Logging class 1"
         consoleOutput shouldContain "Logging class 2"
     }
+
+    @Nested
+    internal inner class `, when given a logging rate,` {
+
+        private val loggingMessage = "logging rate test"
+
+        @Test
+        fun `log on the first invocation`() {
+            val invocations = 3
+            val loggingRate = invocations
+            val consoleOutput = tapConsole {
+                repeat(invocations) {
+                    logger.atInfo()
+                        .every(loggingRate)
+                        .log { loggingMessage }
+                }
+            }
+            consoleOutput shouldContainOnlyOnce loggingMessage
+        }
+
+        @Test
+        fun `use the latest specified rate`() {
+            val invocations = 10
+            val initialLoggingRate = 3
+            val finalLoggingRate = 5
+            val expectedTimesLogged = 2
+            val consoleOutput = tapConsole {
+                repeat(invocations) {
+                    logger.atInfo()
+                        .every(initialLoggingRate)
+                        .every(finalLoggingRate)
+                        .log { loggingMessage }
+                }
+            }
+            val timesLogged = consoleOutput.occurrencesOf(loggingMessage)
+            timesLogged shouldBe expectedTimesLogged
+        }
+
+        @Test
+        fun `throw on zero rate`() {
+            val loggingRate = 0
+            assertThrows<IllegalArgumentException> {
+                logger.atInfo()
+                    .every(loggingRate)
+            }
+        }
+
+        @Test
+        fun `log no more often than the rate allows`() {
+            val loggingRate = 5
+            val invocations = 28
+            val expectedLoggedInvocations = listOf(1, 6, 11, 16, 21, 26)
```

Comment:
Can we please calculate this in code rather than pre-fill? 

In this way, the test will be more reliable, if we, say, want to use it in the future in order to debug a certain behaviour with values other currently hardcoded values of rate and number of invocations.